docs_src/design/adr/security/0019-spiffe.md

```diff
@@ -0,0 +1,158 @@
+# Use SPIFFE/SPIRE for On-demand Secret Store Token Generation
+
+## Status
+
+** DRAFT **
+
+## Context
+
+In security-enabled EdgeX,
+there is a component called `security-secretstore-setup`
+that seeds authentication tokens for Hashicorp Vault--EdgeX's secret store--into
+directories reserved for each EdgeX microservice.
+The implementation is provided by a sub-component, `security-file-token-provider`,
+that works off of a static configuration file and an environment variable
+that lists the services that require tokens
+and the policy to be attached to those tokens.
+
+The current solution has some problematic aspects:
+
+* These tokens have an initial TTL of one hour (1h)
+  and become invalid if not used and renewed within that time period.
+  In practical terms,
+  this means that the EdgeX framework must be shut down and restarted completely
+  in order to change the list of running services.
```

Comment:
In 1.x versions of EdgeX it was possible to restart security-secretstore-setup alone to renew invalid tokens and/or generate tokens for newly added services.